src/VisualStudio/Core/Def/Implementation/ProjectSystem/RuleSets/VisualStudioRuleSetManager.RuleSetFile.cs

```diff
@@ -11,60 +11,72 @@ namespace Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem
 {
     internal sealed partial class VisualStudioRuleSetManager
     {
-        private sealed class RuleSetFile : IRuleSetFile
+        private sealed class RuleSetFile : IRuleSetFile, IDisposable
         {
-            private readonly string _filePath;
-            private readonly List<FileChangeTracker> _trackers;
             private readonly VisualStudioRuleSetManager _ruleSetManager;
+            private readonly object _gate = new object();
+
+            /// <summary>
+            /// The list of file trackers we have. This is null if we haven't even computed trackers yet; but can be empty if we
+            /// already know the file was modified and we unsubscribed.
+            /// </summary>
+            private List<FileChangeTracker> _trackers;
 
             private ReportDiagnostic _generalDiagnosticOption;
             private ImmutableDictionary<string, ReportDiagnostic> _specificDiagnosticOptions;
             private bool _subscribed = false;
             private bool _optionsRead = false;
+            private bool _disposed = false;
```

Comment:
💭 `_disposed` is an anti-pattern for types wrapped in `ReferenceCountedDisposable<T>`.

Reply:
Renamed this and extracted Dispose to a helper for clarity.